pkg/plugins/backendplugin/manager.go

```diff
@@ -231,7 +231,9 @@ func (m *manager) CallResource(config PluginConfig, c *models.ReqContext, path s
 	}
 
 	c.WriteHeader(res.Status)
-	c.Write(res.Body)
+	if _, err := c.Write(res.Body); err != nil {
+		p.logger.Error("Failed to write resource response", "error", err)
```

Comment:
Why log instead of returning error?

Reply:
Because this method handles both request and response so thought it would make most sense to not return an error. This case datasources and plugins in api package don't need to check for error and log it. I'm eager to change this if you want/know a good solution because this situation I'm not sure how to handle, i.e. response status have already been written, but body couldn't be written so the result is an empty response to the client with headers and status set.

Reply:
Btw, before these changes I used the `NormalResponse` and that sort does the same thing, see  https://github.com/grafana/grafana/blob/master/pkg/api/common.go#L59.

Reply:
Research: 
 - https://stackoverflow.com/questions/57828645/how-to-handle-superfluous-response-writeheader-call-in-order-to-return-500 (Says log it)

Also https://github.com/prometheus/prometheus/issues/4307#issuecomment-400653086 explains the scenario well.

So as I understand it, basically the common case would be that the connection gets closed while / before writing the body. In which case the error won't make it back anyways. 

Therefore I agree, logging is correct.

Reply:
Thanks. Please approve PR then :smile: